app/src/main/java/org/eyeseetea/malariacare/domain/usecase/pull/PullDemoUseCase.java

```diff
@@ -31,22 +35,51 @@
     }
 
     IPullDemoController mPullController;
+    private IAsyncExecutor mAsyncExecutor;
+    private IMainExecutor mMainExecutor;
+    private Callback mCallback;
 
-    public PullDemoUseCase(IPullDemoController pullController) {
+    public PullDemoUseCase(IPullDemoController pullController, IMainExecutor mainExecutor,
+            IAsyncExecutor asyncExecutor) {
         mPullController = pullController;
+        mMainExecutor = mainExecutor;
+        mAsyncExecutor = asyncExecutor;
     }
 
-    public void execute(final Callback callback) {
+    @Override
+    public void run() {
         mPullController.pull(new PullDemoController.IPullDemoControllerCallback() {
-
             @Override
             public void onComplete() {
-                callback.onComplete();
+                notifyOnComplete();
             }
 
             @Override
             public void onError(Throwable throwable) {
-                callback.onPullError();
+                notifyOnPullError();
+            }
+        });
+    }
+
+    public void execute(final Callback callback) {
```

Comment:
Please move execute method before run method for we have a code more readable according to execution flow